### ISSUE TITLE
Safer event handling on native side

### ISSUE THREAD

**Issue by maicki** [maicki/react-native-event-bridge#12]
The native side does not verify if events that are happening from react native side are supported or not. We should add a way to verify that the component managed by the event receiver is sending only events up that are supported by the event receiver.

Some ideas:
1. On the native side we should added to the receiver protocol:
```
/**
 * Return an array of supported events.
 */
- (NSArray *)supportedEvents;
```

And in `onEvent:` callbacks check if the event that was sent from react native is 

2. Change the method of the `onEvent:` callbacks to return a boolean if the event was handled and in case the event is returned as a supported event but not handle assert in dev mode:
```
/**
 * Handle event received from React Native. Return YES if the event will be handled, otherwise if the event will not
 * be handled an assertion will happen in development.
 */
- (BOOL)onEventWithName:(NSString *)eventName info:(nullable NSDictionary *)info
```

**Comment by maicki** [comment]
The approach from this issues is implemented on iOS in #13 but not on Android yet. It should not be so hard to get this changes over. If someone would like to pick it up please let me know.

**Comment by maicki** [comment]
Closing this PR. For further information please see: #48
